common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -319,7 +319,7 @@ public static void throwException(Throwable t) {
   static {
     boolean _unaligned;
     String arch = System.getProperty("os.arch", "");
-    if (arch.equals("ppc64le") || arch.equals("ppc64") || arch.equals("s390x")) {
+    if ("ppc64le".equals(arch) || "ppc64".equals(arch) || "s390x".equals(arch)) {
```

Comment:
In what scenario`System.getProperty("os.arch", ""` is null?

Reply:
no scenario. back off